pkgs/applications/audio/faust/faust2alqt.nix

```diff
@@ -1,15 +1,39 @@
 { faust
 , alsa-lib
-, qt4
+, qtbase
+, writeText
+, makeWrapper
 }:
 
-faust.wrapWithBuildEnv {
+faust.wrapWithBuildEnv rec {
 
   baseName = "faust2alqt";
 
   propagatedBuildInputs = [
     alsa-lib
-    qt4
+    qtbase
   ];
 
+  dontWrapQtApps = true;
+
+  # Wrap the binary coming out of the the compilation script, so it knows QT_PLUGIN_PATH
+  wrapBinary = writeText "wrapBinary" ''
+    source ${makeWrapper}/nix-support/setup-hook
+    for p in $FILES; do
+      binary=$(basename --suffix=.dsp "$p")
+      mv $binary ."$binary"-wrapped
+      makeWrapper ./."$binary"-wrapped ."$binary"-temp  --set QT_PLUGIN_PATH "${qtbase}/${qtbase.qtPluginPrefix}"
+      mv ."$binary"-temp $binary
+    done
+  '';
```

Comment:
Isn't this just `wrapQtAppsHook`?

Reply:
I think we had some trouble getting that to work in #180387, but it probably is, yes.

Reply:
``wrapQtAppsHook`` would wrap the faust compiler, we want to wrap the output of the compiler.

Reply:
Is the compiler something that runs at _runtime_? So this package is just the compiler, and it's its outputs that need wrapping?

Reply:
Yes.

Reply:
I fixed the directory issue.